packages/react/src/components/header/components/headerLanguageSelector/HeaderLanguageSelector.tsx

```diff
@@ -1,35 +1,25 @@
-import React, { PropsWithChildren, useEffect } from 'react';
+import React, { PropsWithChildren } from 'react';
 
-import {
-  LanguageOption,
-  useAvailableLanguages,
-  useActiveLanguage,
-  useSetAvailableLanguages,
-  useSetLanguage,
-} from '../../LanguageContext';
+import { LanguageOption, useActiveLanguage, useAvailableLanguages, useSetLanguage } from '../../LanguageContext';
 import classNames from '../../../../utils/classNames';
-import { getChildElementsEvenIfContainersInbetween } from '../../../../utils/getChildren';
 import { withDefaultPrevented } from '../../../../utils/useCallback';
 import { HeaderActionBarItemWithDropdown } from '../headerActionBarItem';
 import { IconAngleDown, IconAngleUp, IconGlobe } from '../../../../icons';
 import classes from './HeaderLanguageSelector.module.scss';
-import { useHeaderContext, useSetHeaderContext } from '../../HeaderContext';
+import { useHeaderContext } from '../../HeaderContext';
+import { getComponentFromChildren } from '../../../../utils/getChildren';
 
-export type LanguageSelectorProps = PropsWithChildren<{
-  /**
-   * Array of languages as LanguageOption objects.
-   */
-  languages?: LanguageOption[];
+type LanguageSelectorComponentProps = {
   /**
    * Aria-label attribute for the dropdown button.
    */
+  // eslint-disable-next-line react/no-unused-prop-types
   ariaLabel?: string;
-  /**
-   * Mobile mode.
-   * @internal
-   */
+  // eslint-disable-next-line react/no-unused-prop-types
   fullWidthForMobile?: boolean;
```

Comment:
Not internal anymore? Then this is suggested to users but I don't see it used in Header.stories. And it would be better if it's automatically used and users don't need to know about it.

Reply:
Changed back to internal.